google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GrpcSpannerRpc.java

```diff
@@ -130,18 +125,19 @@ public GrpcSpannerRpc(SpannerOptions options) {
       stubsBuilder.add(withCredentials(SpannerGrpc.newFutureStub(channel), credentials));
     }
     this.channels = channelsBuilder.build();
-    TransportOptions transportOptions = options.getTransportOptions();
-    // Note, getXGoogApiClientHeader() method can be added to TransportOptions directly.
-    // Doing explicit casting here (instead of "contaminating" the top level interface) since most
-    // probably it is a temporary solution (eventually all grpc clients will become gapic-based,
-    // and gapic handles the header internally).
-    if (transportOptions instanceof GrpcTransportOptions) {
-      this.xGoogApiClientHeader =
-          ((GrpcTransportOptions) transportOptions)
-              .getXGoogApiClientHeader(options.getLibraryVersion());
-    } else {
-      this.xGoogApiClientHeader = "";
-    }
+
+    ApiClientHeaderProvider.Builder internalHeaderProviderBuilder
+        = GrpcClientHeaderProvider.newBuilder();
+    ApiClientHeaderProvider internalHeaderProvider =
+        internalHeaderProviderBuilder
+            .setClientLibToken(
+                ServiceOptions.getGoogApiClientLibName(),
+                GaxProperties.getLibraryVersion(options.getClass()))
+            .build();
+
+    HeaderProvider mergedHeaderProvider = options.getMergedHeaderProvider(internalHeaderProvider);
+    this.headerProvider = SpannerHeaderProvider.create(mergedHeaderProvider.getHeaders(),
+        internalHeaderProviderBuilder.getResourceHeaderKey());
```

Comment:
This key is a constant so I should be able to get to it by referring to a static constant rather than as a member method on ApiClientHeaderProvider. Ideally there should be no need to pass it to SpannerHeaderProvider. It should just refer to the constant somewhere.

Reply:
On gapic level this value is treated as the one with default value, but not constant. In this particular case, the value keeps the default value.